### PR TITLE
[growatt] Enhancements for MID inverter and Smart Meter

### PR DESCRIPTION
**Pull request by andrewfg**
See #17799

Signed-off-by: AndrewFG <software@whitebear.ch>

### CODE REVIEW

File: bundles/org.openhab.binding.growatt/src/main/resources/OH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -480,10 +480,99 @@
...
+			</channel>
+			<channel id="pv4-power" typeId="advanced-electric-power">
+				<label>String #4 Power</label>
+				<description>Power from solar panel string #4.</description>
+			</channel>
+
+			<!-- solar DC pv energy -->
+			<channel id="pv3-energy-today" typeId="advanced-electric-energy">
+				<label>DC Energy #3 Today</label>
+				<description>Solar DC energy collected by string #3.</description>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.growatt/src/main/resources/OH-INF/thing/thing-types.xml:515]: Maybe "... collected **today** by ..."

File: bundles/org.openhab.binding.growatt/src/main/resources/OH-INF/update/instructions.xml [hunk 2]
```diff
@@ -49,6 +49,107 @@
...
+			</add-channel>
+			<add-channel id="pv4-power">
+				<type>growatt:advanced-electric-power</type>
+				<label>String #4 Power</label>
+				<description>Power from solar panel string #4.</description>
+			</add-channel>
+			<add-channel id="pv3-energy-today">
+				<type>growatt:advanced-electric-energy</type>
+				<label>DC Energy #3 Today</label>
+				<description>Solar DC energy collected by string #3.</description>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.growatt/src/main/resources/OH-INF/update/instructions.xml:89]: To be adjusted if previous suggestion is considered

File: bundles/org.openhab.binding.growatt/src/main/resources/OH-INF/thing/thing-types.xml (hunk 1 above)
Comment (andrewfg) [reply bundles/org.openhab.binding.growatt/src/main/resources/OH-INF/thing/thing-types.xml]: Done.

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**